Port set_ to ATen.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34403 Port set_ to ATen.**

Differential Revision: [D20342300](https://our.internmc.facebook.com/intern/diff/D20342300)

gchanan committed:
Port set_ to ATen.

gchanan committed:
Update on "Port set_ to ATen."


Differential Revision: [D20342300](https://our.internmc.facebook.com/intern/diff/D20342300)

gchanan committed:
Update on "Port set_ to ATen."


Differential Revision: [D20342300](https://our.internmc.facebook.com/intern/diff/D20342300)

gchanan committed:
Update on "Port set_ to ATen."


Differential Revision: [D20342300](https://our.internmc.facebook.com/intern/diff/D20342300)

gchanan committed:
Update on "Port set_ to ATen."


Differential Revision: [D20342300](https://our.internmc.facebook.com/intern/diff/D20342300)

gchanan committed:
Update on "Port set_ to ATen."


Differential Revision: [D20342300](https://our.internmc.facebook.com/intern/diff/D20342300)

gchanan committed:
Update on "Port set_ to ATen."


Differential Revision: [D20342300](https://our.internmc.facebook.com/intern/diff/D20342300)

gchanan committed:
Update on "Port set_ to ATen."


Differential Revision: [D20342300](https://our.internmc.facebook.com/intern/diff/D20342300)

gchanan committed:
Update on "Port set_ to ATen."


Differential Revision: [D20342300](https://our.internmc.facebook.com/intern/diff/D20342300)

gchanan committed:
Update on "Port set_ to ATen."


Differential Revision: [D20342300](https://our.internmc.facebook.com/intern/diff/D20342300)

gchanan committed:
Update on "Port set_ to ATen."


Differential Revision: [D20342300](https://our.internmc.facebook.com/intern/diff/D20342300)

bhosmer commented on aten/src/ATen/native/TensorShape.cpp:
```diff
@@ -42,6 +42,50 @@ Tensor& set_(Tensor& result, Storage source) {
   return result.set_(source, 0, static_cast<int64_t>(source.size()), {});
 }
 
+// unify with cuda implementation?  This is not done to avoid a dispatch in resize_impl_cpu_
```
It looks like big stretches of the error checking etc. are identical between these two - would it be a hassle to factor that stuff out into a common helper?

gchanan replied:
I think that would work, let me try.

bhosmer commented on aten/src/TH/THTensor.cpp:
```diff
@@ -25,47 +25,8 @@ void THTensor_free(THTensor *self)
 }
 
 void THTensor_setStorage(THTensor *self, THStorage *storage_, ptrdiff_t storageOffset_, at::IntArrayRef size_, at::IntArrayRef stride_) {
```
So after this PR, who calls this (resp. the cuda version)? Or is it part of the public API and never going away?

gchanan replied:
good question.  It's all internal calls, i.e. things calling `THTensor_(set)`, `THTensor_(newWithStorage1d)`.  Most of those look easy to remove except that the copy code uses `THTensor_(newWithStorage1d)` and that's a bit hairy.

bhosmer replied:
Thanks for the info - yeah just looked at the callsites of `newWithStorageId` and they look nontrivial to excise :)

bhosmer reviewed (approved):
Nice legacy prune :) it all checks out to the extent I know what's going on, but LMK if there's any subtleties worth digging into. One suggestion inline but totally your call.

gchanan committed:
Update on "Port set_ to ATen."


Differential Revision: [D20342300](https://our.internmc.facebook.com/intern/diff/D20342300)

gchanan committed:
Update on "Port set_ to ATen."


Differential Revision: [D20342300](https://our.internmc.facebook.com/intern/diff/D20342300)
